### PR TITLE
Preview logic requires tweaking

### LINKED ISSUE #14968: [Drag & Drop] Preview logic requires tweaking.
## 📝 Provide detailed reproduction steps (if any)

Fix the logic:

1.  When selected above the block toolbar ⇒ above the editable.
    1.  This works now, we may check if the implementation in the experimental is correct.
2.  When in the editable, show preview near the cursor.

Otherwise, in some cases, the preview won't be visible.

https://github.com/ckeditor/ckeditor5/assets/9881379/2833fea6-50b6-44cf-a7af-3abf3b4389c6

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by illia-stv**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (clipboard): improve displaying preview in "drag and drop". Closes #14968.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Review by niegowski — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-clipboard/src/dragdropexperimental.ts [hunk 1]
```diff
@@ -35,7 +35,8 @@ import {
 	delay,
 	type DelayedFunc,
 	type ObservableChangeEvent,
-	type DomEmitter
+	type DomEmitter,
+	Rect
```
Comment (filipsobol) [inline packages/ckeditor5-clipboard/src/dragdropexperimental.ts:39]: Regular imports should go before `type` imports

File: packages/ckeditor5-clipboard/src/dragdropexperimental.ts [hunk 2]
```diff
@@ -632,18 +644,27 @@ export default class DragDropExperimental extends Plugin {
...
 			global.document.body.appendChild( this._previewContainer );
-		} else {
-			this._previewContainer.removeChild( this._previewContainer.firstElementChild! );
+		} else if ( this._previewContainer.firstElementChild ) {
+			this._previewContainer.removeChild( this._previewContainer.firstElementChild );
+		}
+
+		const domRect = new Rect( domEditable );
+
+		// If domTarget is inside editable root, browser display preview by own.
```
Comment (filipsobol) [inline packages/ckeditor5-clipboard/src/dragdropexperimental.ts:653]:
```suggestion
		// If domTarget is inside the editable root, browsers will display the preview correctly by themselves.
```

I think this reads better.

File: packages/ckeditor5-clipboard/tests/dragdropexperimental.js [hunk 3]
```diff
@@ -2154,13 +2240,17 @@ describe( 'Drag and Drop experimental', () => {
...
+					domConverter.findCorrespondingDomText( viewPosition.parent ).parentNode :
+					domConverter.mapViewToDom( viewElement );
+			} else {
+				domNode = domTarget;
+			}
 		} else {
 			viewRange = mapper.toViewRange( modelPositionOrRange );
 			viewElement = viewRange.getContainedElement();
-			domNode = domConverter.mapViewToDom( viewElement );
+			domNode = !domTarget ? domConverter.mapViewToDom( viewElement ) : domTarget;
```
Comment (filipsobol) [inline packages/ckeditor5-clipboard/tests/dragdropexperimental.js:2253]: domNode = domTarget || domConverter.mapViewToDom( viewElement );

Comment (filipsobol) [inline packages/ckeditor5-clipboard/tests/dragdropexperimental.js:2253]:
```suggestion
			domNode = domTarget || domConverter.mapViewToDom( viewElement );
```

File: packages/ckeditor5-clipboard/tests/dragdropexperimental.js [hunk 4]
```diff
@@ -1442,6 +1446,66 @@ describe( 'Drag and Drop experimental', () => {
 				// There should be two elements with the `.ck-content` class - editor and drag-and-drop preview.
 				expect( numberOfCkContentElements ).to.equal( 2 );
 			} );
+
+			it( 'should show preview with custom implementation', () => {
```
Comment (filipsobol) [inline packages/ckeditor5-clipboard/tests/dragdropexperimental.js:1450]: It's not clear from the test names alone when the editor `should show preview with custom implementation` and when it `should show preview with browser implementation`. Please include information when one or the other is used, so it's not necessary to read the test details.

For example: `should show preview with custom implementation if [...]` and `should show preview with browser implementation if [...]`.

**Review by filipsobol — COMMENTED** [review]
I added few minor comments

**Review by niegowski — APPROVED** [review]

**Pull request merged**